src/SFML.Audio/SoundBufferRecorder.cs

```diff
@@ -57,9 +58,14 @@ protected override bool OnStart()
         /// <param name="samples">Array of samples to process</param>
         /// <returns>False to stop recording audio data, true to continue</returns>
         ////////////////////////////////////////////////////////////
-        protected override bool OnProcessSamples(short[] samples)
+        protected override bool OnProcessSamples(Span<short> samples)
         {
-            mySamplesArray.AddRange(samples);
+            if (mySamplesArray.Capacity < mySamplesArray.Count + samples.Length)
+                mySamplesArray.Capacity = mySamplesArray.Count + samples.Length;
+
+            for (int i = 0; i < samples.Length; i++)
```

Comment:
Is there not a way that we can use `AddRange()` with `Span`?  Also, Audio is likely going to benefit from a significant rework, but that's beyond the scope of this draft.

Reply:
There is no such overload for `AddRange()` if I remember correctly.